packer/sources.pkr.hcl

```diff
@@ -1,28 +1,37 @@
 source "amazon-ebs" "builder" {
-  ami_name              = "${var.ami_name_prefix}-${var.version}"
-  ami_users             = var.ami_account_ids
-  communicator          = "ssh"
-  force_delete_snapshot = var.force_delete_snapshot
-  force_deregister      = var.force_deregister
-  instance_type         = var.aws_instance_type
-  region                = var.aws_region
-  ssh_private_key_file  = var.ssh_private_key_file
-  ssh_username          = var.ssh_username
-  ssh_keypair_name      = "packer-builders-${var.aws_region}"
-  iam_instance_profile  = "packer-builders-${var.aws_region}"
+  ami_name                  = "${var.ami_name_prefix}-${var.version}"
+  ami_users                 = var.ami_account_ids
+  communicator              = "ssh"
+  force_delete_snapshot     = var.force_delete_snapshot
+  force_deregister          = var.force_deregister
+  instance_type             = var.aws_instance_type
+  region                    = var.aws_region
+  ssh_clear_authorized_keys = var.ssh_clear_authorized_keys
+  ssh_private_key_file      = var.ssh_private_key_file
+  ssh_username              = var.ssh_username
+  ssh_keypair_name          = "packer-builders-${var.aws_region}"
+  iam_instance_profile      = "packer-builders-${var.aws_region}"
 
   launch_block_device_mappings {
-    device_name = "/dev/sda1"
-    volume_size = var.root_volume_size_gb
-    volume_type = "gp2"
     delete_on_termination = true
+    device_name           = "/dev/sda1"
+    encrypted             = true
+    iops                  = var.root_volume_iops
+    kms_key_id            = var.kms_key_id
+    throughput            = var.root_volume_throughput
+    volume_size           = var.root_volume_size_gb
+    volume_type           = "gp3"
```

Comment:
While I can't see this changing any time soon is it worth specifying this as a variable with a `gp3` default?

Reply:
It seems like the sensible thing to do, but the `throughput` argument is only valid for `gp3` types and introducing a variable here might cause the configuration to become invalid if changed from the default value of `gp3`. If the variable value can't be changed it would seem to defeat the purpose of it. The situation could change of course, if additional volume types are introduced, but then the variable would only support a subset of the available types, which we'd have to document , and that may itself introduce the need to use `dynamic` blocks if the configuration for each type we support needs to differ.

Given some of the improvements that the `gp3` volume type introduces, such as predictable baseline performance that isn't linked to volume size, and lower cost per GB compared to `gp2`, enforcing it as the default here seems like a sensible approach for new projects. Other volume types can still be used of course, but that's an exercise for the project author, to build on the template configuration provided in this project.